### PR TITLE
Summary of Commits for Starting a New Game

### PR DESCRIPTION
**Pull request by GrandSuccess87**
All test cases in test suite pass. I deleted source code from my
components that I hadn't tested first and started over so that this time
I implemented TDD first before writing my source code. Modified the
source code accordingly as I worked through the green/red TDD cycle. The
empty board is now created and renders to a new page at the /board url.
Heading for the players info renders at the /game url. Currently working
on a way to render both of these components to the /game url. Also
modified the header component so that the text for player 1 and player 2
is evenly distributed across 2 columns.

Second story is completed. Another page is shown when the start game is
clicked. On the second page the user is able to see an empty tic
tac toe board in which the spaces on the board are simply div tags
styled with css. The heading for the players info is also shown on the same
page. All tests in the test suite pass. There are currently some tests
which test static components. I will remove these if it's decided that
they should removed. All tests in the test suite pass. Also created a game component that encompasses all of the components for the app.

Today I also modified the test suite to include two more tests which
will find  the css class selector of board-grid for the board component
and the css id selector of space for the square component. All tests in test suite  pass.

updated working tree app.test.js

### CODE REVIEW

File: src/pages/game/game.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import React, { Component } from 'react';
-import GameHeader from '../../components/gameHeaderComponent/gameHeaderComponent';
+import Game from '../../components/boardComponent/boardComponent';
 
 class gamePage extends Component {
     render() {
+        console.log('page 2 loaded')
```
Comment (soft-circles) [inline src/pages/game/game.js]: Remember to get rid of any `console.logs` before pushing up.

File: src/components/boardComponent/boardComponent.js [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
-                    <div className="row">
-                        <div className="board-grid col-lg-4 col-md-4 col-sm-4 col-xs-4">
-                            <Square />
-                            <Square />
-                            <Square />
+            <div> 
+                <GameHeader />
+                <div className="Board Board_Row">
+                        <div className="text instructions-text">
+                            <div className="row">
```
Comment (soft-circles) [inline src/components/boardComponent/boardComponent.js]: I'm confused why the `className` for this `div` is `row`. A `row` contains a `board-grid`?

File: src/App.test.js [hunk 3]
```diff
@@ -2,27 +2,71 @@ import React from 'react';
...
+    ReactDOM.unmountComponentAtNode(boardDiv);
+  });
+  
+  it('renders the square component without crashing', () => {
+    const gameSpaceDiv = document.createElement('div');
+    ReactDOM.render(<GameSquare />, gameSpaceDiv);
+    ReactDOM.unmountComponentAtNode(gameSpaceDiv);
+  });
+
+  it('renders one <Square /> component within the <Board /> componenet without crashing', () => {
```
Comment (soft-circles) [inline src/App.test.js]: you can just say "renders one Square component within the Board component without crashing"

**Review by soft-circles — CHANGES_REQUESTED** [review]
Please separate your tests in a file structure that mirrors your application. Have your `Board` class tests live in `tests/components/BoardTests` or something like that. I'm not sure what the file naming  conventions are in React but it's best to group tests together so they're easier to reason about and find.

File: src/pages/game/game.js (hunk 1 above)
Comment (GrandSuccess87) [reply src/pages/game/game.js]: Thought I had got them all, just removed this one

File: src/components/boardComponent/boardComponent.js (hunk 2 above)
Comment (GrandSuccess87) [reply src/components/boardComponent/boardComponent.js]: It shouldn't be row, I can see why that's confusing. I initially used rows to build the board. Just changed the naming so it makes more sense.

File: src/App.test.js (hunk 3 above)
Comment (GrandSuccess87) [reply src/App.test.js]: Thanks Andy, just modified my tests to mimic this

**Comment by GrandSuccess87** [comment]
Thanks for the feedback Andy! I separated my tests into a more clean and organized file structure, should be easier for everyone to read now.

File: src/components/squareComponent/squareComponent.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
+import './squareComp.css'
+
+class Square extends Component {
+
+    render() {
+        return (
+            <div>
+                <a href="/"><div className="game-spaces" id="space"></div></a>
```
Comment (soft-circles) [inline src/components/squareComponent/squareComponent.js]: Why is this a `<a>` tag? Could this just be a an unstyled `button` or just a `div`?

Comment (GrandSuccess87) [reply src/components/squareComponent/squareComponent.js]: Yea I was just thinking on this as I'm going over my code.  This isn't how I should do it. I'm going to create an onClick function that will log to the console the space has been clicked. Is that okay?

File: src/components/squareComponent/square.spec.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        const wrapper = shallow(<GameBoard />);
+        const boardGridDiv = wrapper.find(GameSquare);
+        expect(wrapper.contains(<GameSquare />)).toBe(true);
+      });
+
+      it('tests that each space on the empty board is clickable', () => {
+        const wrapper = shallow(<GameSquare />);
+        const boardDiv = wrapper.find(GameSquare);
+        const clickSquare = wrapper.find('a').simulate('click');
+        expect(clickSquare.length).toEqual(1);
```
Comment (soft-circles) [inline src/components/squareComponent/square.spec.js]: how useful is this test? Does anything actually happen when that 'a' is clicked?

Comment (GrandSuccess87) [reply src/components/squareComponent/square.spec.js]: I suppose not too useful. Especially if I'm removing the a tag.

File: src/components/squareComponent/squareComponent.js (hunk 4 above)
Comment (GrandSuccess87) [reply src/components/squareComponent/squareComponent.js]: A tag is removed, I've created a function that now handles the click event

File: src/components/squareComponent/square.spec.js (hunk 5 above)
Comment (ericmeyer) [reply src/components/squareComponent/square.spec.js]: 👍 to @akiraandy 

It's usually best to hold off implementing features you don't need yet. In this case, that includes making squares clickable.

Comment (GrandSuccess87) [reply src/components/squareComponent/square.spec.js]: Okay, got it, I can remove this feature and hold off on it for now

**Pull request merged**